packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -279,7 +279,15 @@ function s(
                         // to the vnode because the current way the diffing algo works, it will replace the original reference
                         // to the host element with a new one. This means the new element will be mounted and immediately unmounted.
                         // Creating a copy of the vnode to preserve a reference to the previous host element.
-                        clonedVNode = { ...vnode, slotAssignment: data.slotAssignment };
+                        if (!isUndefined(vnode.elm)) {
```

Comment:
I would prefer to reverse the logic here. `if (!foo) {} else {}` is always harder to read than `if (foo) else {}`.